shared/src/main/kotlin/gov/cdc/prime/reportstream/shared/QueueMessage.kt

```diff
@@ -0,0 +1,196 @@
+         *
+         * @param s Base64 encoded string representing the message.
+         * @return Deserialized gov.cdc.prime.reportstream.shared.QueueMessage object.
+         */
+        fun deserialize(s: String): QueueMessage = mapper.readValue(s)
+
+        /**
+         * Returns a JSON string representation of the gov.cdc.prime.reportstream.shared.QueueMessage object.
+         *
+         * @return JSON string representation.
+         */
+        override fun toString(): String = mapper.writeValueAsString(this)
+
+        /**
+         * Constant for receive queue on UP
+         */
+        const val elrReceiveQueueName = "elr-fhir-receive"
+
+        /**
+         * Constant for convert queue on UP
+         */
+        const val elrConvertQueueName = "elr-fhir-convert"
+
+        /**
+         * Constant for destination filter queue on UP
+         */
+        const val elrDestinationFilterQueueName = "elr-fhir-destination-filter"
+
+        /**
+         * Constant for receiver filter queue on UP
+         */
+        const val elrReceiverFilterQueueName = "elr-fhir-receiver-filter"
+
+        /**
+         * Constant for translation queue on UP
+         */
+        const val elrTranslationQueueName = "elr-fhir-translate"
+
+        /**
+         * Constant for send queue
+         */
+        const val elrSendQueueName = "send"
+    }
+
+    /**
+     * Interface representing information about a report to be processed.
+     */
+    interface ReportInformation {
+        /**
+         * URL of the blob storage containing the report.
+         */
+        val blobURL: String
+
+        /**
+         * Digest (hash) of the report for integrity verification.
+         */
+        val digest: String
+
+        /**
+         * Subfolder name in the blob storage where the report is stored.
+         */
+        val blobSubFolderName: String
+
+        /**
+         * Unique identifier of the report.
+         */
+        val reportId: UUID
+    }
+
+    /**
+     * Interface representing additional information required for receiving a message.
+     */
+    interface ReceiveInformation {
+        /**
+         * Additional headers associated with the message.
+         */
+        val headers: Map<String, String>
+    }
+
+    /**
+     * Data class representing a specific type of gov.cdc.prime.reportstream.shared.QueueMessage meant for receiving
+     * FHIR (Fast Healthcare Interoperability Resources) data. It implements both
+     * ReportInformation and ReceiveInformation interfaces.
+     *
+     * @property blobURL The URL of the blob storage containing the report.
+     * @property digest The digest (hash) of the report.
+     * @property blobSubFolderName The subfolder name in the blob storage.
+     * @property reportId The unique identifier of the report.
+     * @property headers Additional headers associated with the message.
+     */
+    @JsonTypeName("receive-fhir")
+    data class ReceiveQueueMessage(
```

Comment:
let's discuss: is there a way to just use this one and not have a mirror of it in PrimeRouterQueueMessage? I assume there is going to be a ticket for this issue so will want to see that to better understand.

Reply:
I think we figured this out in a different comment but just in case, we need them both since they are used in different modules. The submission controller creates `ReceiveQueueMessage` but the UP can only work with `PrimeRouterQueueMessage`. So the FHIRReceiveQueueMessage is needed there. I did add a ticket to move all queue message classes to `QueueMessage` though.